tests/settings/SectionTest.py

```diff
@@ -22,6 +23,24 @@ def test_append(self):
         self.assertEqual(str(uut.get('5 ')), '5')
         self.assertEqual(int(uut.get('nonexistent', 5)), 5)
 
+    def test_settings_override_warning(self):
+        logger = logging.getLogger()
+        with self.assertLogs(logger, 'WARNING') as cm:
+            uut = Section('name', None)
+            uut.add_or_create_setting(Setting('key1', 'value1'), None, False)
+            uut.add_or_create_setting(Setting('key1', 'value2'), None, False)
+        self.assertEqual(cm.output[0], 'WARNING:root:key1 setting has already '
+                                       'been defined in section name. The '
+                                       'previous setting will be overridden.')
+        with self.assertLogs(logger, 'WARNING') as cm:
+            uut = Section('name', None)
+            uut.add_or_create_setting(Setting('key2', 'value1'), None, False)
+            uut.add_or_create_setting(Setting('key2', 'value2'), None, True)
```

Comment:
Since we parse each line for the settings, the first time we parse a setting, `allow_appending` will be False. Now suppose our setting spans multiple lines, and when we parse the second line, `allow_appending` will be set to `True` and no warning will be issued. If the setting is encountered again, `allow_appending` will be `False` and a warning will be issued if the setting is already defined(checking `self.contents.keys()` for already defined settings). Is that okay @aptrishu ?